sphinx_gherkindoc/writer.py

```diff
@@ -315,14 +317,24 @@ def steps(steps: List[behave.model.Step], step_format: str = "{}") -> None:
                 text(step.text)
 
     def examples(
-        example_source: Union[behave.model.Scenario, behave.model.Feature],
+        example_source: Union[
+            behave_models.Scenario,
+            pytest_bdd_models.Scenario,
+            pytest_bdd_models.Feature,
```

Comment:
This is surprising, I expected to see `pytest_bdd_models.Feature` added to the `Union` but I did not expect to see `behave_models.Feature` removed...

Reply:
I'd like to understand this change before signing off.

Reply:
So the short of it is that the current type hints are actually off. Since we define our own models for each parser ourselves, the original `behave.model.Scenario, behave.model.Feature` actually would never be the type of `example_source`. Since it is related to these changes (the isinstance check), I made it correct here. `behave_models.Feature` can never be passed in here because it never can have a `.feature` value. The line that calls this checks first `if feature.examples:`.

I'll have a FF to clean up all of the type hints in this file to use the correct `behave_models` I added instead of the `behave.model` values, which are now inaccurate. You'll notice that no where else in the type hints are the pytest_bdd models mentioned, so they need to be everywhere really. I'll try to organize it concisely and clearly.

Reply:
Ok, so basically the type hinting is wrogn and will be fixed in another PR.

Reply:
Uf, I hate that I (and `mypy`?) missed this in the midst of the larger work.